src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -22,20 +22,39 @@ public SpectralManager(ILoggerWrapper loggerWrapper, string openApiToolsDirector
     }
     
     private string ExecutablePath { get; set; } = string.Empty;
+    
+    private bool SkipRun { get; set; }
 
     public async Task InstallSpectralAsync(CancellationToken cancellationToken)
     {
-        Directory.CreateDirectory(this._spectralDirectory);
+        try
+        {
+            this._loggerWrapper.LogMessage("Starting Spectral installation.");
+            
+            Directory.CreateDirectory(this._spectralDirectory);
 
-        this.ExecutablePath = GetSpectralFileName();
-        var url = $"https://github.com/stoplightio/spectral/releases/download/v{SpectralVersion}/{this.ExecutablePath}";
-        var destination = Path.Combine(this._spectralDirectory, this.ExecutablePath);
+            this.ExecutablePath = GetSpectralFileName();
+            var url = $"https://github.com/stoplightio/spectral/releases/download/v{SpectralVersion}/{this.ExecutablePath}";
+            var destination = Path.Combine(this._spectralDirectory, this.ExecutablePath);
         
-        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, destination, cancellationToken);
+            await this._httpClientWrapper.DownloadFileToDestinationAsync(url, destination, cancellationToken);
+            
+            this._loggerWrapper.LogMessage("Spectral installation completed.");
+        }
+        catch (Exception)
+        {
+            this.SkipRun = true;
+            throw;
+        }
     }
 
     public async Task RunSpectralAsync(IEnumerable<string> swaggerDocumentPaths, string rulesetUrl, CancellationToken cancellationToken)
```

Comment:
I tried to make Spectral validation fail to generate an error report by adding an enum field with duplicate entries in my YAML spec but still Spectral says that everything looks good. Any idea how? And also do you have an example of a spec that would fail the validation?

![image](https://github.com/gsoft-inc/wl-openapi-msbuild/assets/14242083/69a59c89-81e5-4575-b0f4-e0108b973a3b)

Reply:
I will investigate this.